src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -478,7 +463,7 @@ internal CompilationContext CreateCompilationContext(CSharpSyntaxNode syntax)
                 }
 
                 ImmutableArray<bool> dynamicFlags;
-                if (dynamicLocalMap.TryGetValue(i, out dynamicFlags))
+                if (dynamicLocalMap != null && dynamicLocalMap.TryGetValue(i, out dynamicFlags))
```

Comment:
Consider requiring the caller to pass non-null dynamicLocalMap, to simplify this method and the method below.